src/daos_vol.c

```diff
@@ -1864,6 +1870,35 @@ H5_daos_get_conn_cls(void *item, H5VL_get_conn_lvl_t H5VL_DAOS_UNUSED lvl,
     D_FUNC_LEAVE_API;
 } /* end H5_daos_get_conn_cls() */
 
+
+/*---------------------------------------------------------------------------
+ * Function:    H5_daos_get_cap_flags
+ *
+ * Purpose:     Retrieves the capability flags for this VOL connector.
+ *
+ * Return:      Non-negative on Success/Negative on failure
+ *
+ *---------------------------------------------------------------------------
+ */
+static herr_t
+H5_daos_get_cap_flags(const void H5VL_DAOS_UNUSED *info, unsigned *cap_flags)
+{
+    herr_t ret_value = SUCCEED;
+
+    H5_daos_inc_api_cnt();
+
+    if(!cap_flags)
+        D_GOTO_ERROR(H5E_ARGS, H5E_BADVALUE, FAIL, "invalid cap_flags parameter");
+
+    H5_DAOS_MAKE_ASYNC_PROGRESS(FAIL);
```

Comment:
would not hurt to have a one line comment about what that means

Reply:
hmm alright... still kind of clunky here but ok I guess...

Reply:
Let's leave it like that for now but maybe put a todo note. I feel like we should review where that gets done in general, I don't think we should allow for progress to be made everywhere otherwise this kind of routine might take longer whereas it is supposed to be immediate.

Reply:
Well it is still immediate in the sense that it never blocks. It just might take an order or two magnitude more CPU time than expected.